lib/src/tree_cursor.h

```diff
@@ -9,6 +9,7 @@ typedef struct {
   uint32_t child_index;
   uint32_t structural_child_index;
   uint32_t descendant_index;
+  TSSymbol root_alias;
```

Comment:
What do you think of putting this on the `TreeCursor` instead of on each node of the stack? I think it would make it clearer that it is only meaningful for the root node.

Reply:
Yeah this makes a lot of sense, it's only relevant to whatever node the cursor was set with